plugins/grounded/grounded.cpp

```diff
@@ -0,0 +1,177 @@
+#include "ProcessWindows.h"
+#define MUMBLE_PLUGIN_NO_DEFAULT_FUNCTION_DEFINITIONS
+#include "MumblePlugin.h"
+#undef MUMBLE_PLUGIN_NO_DEFAULT_FUNCTION_DEFINITIONS
+
+#include "mumble_positional_audio_utils.h"
+
+
+using GroundedHandle = std::tuple< ProcessWindows, procptr_t >;
+
+static std::unique_ptr< GroundedHandle > handle;
+
+/* the indexes of these float[3] from the game are
+ *   0: south low, north high
+ *   1: west low, east high
+ *   2: altitude, high toward sky, goes up when you jump */
+struct GroundedCam {
+	/* 940 */ float top[3];
+	uint8_t _unused1[4];
+	/* 950 */ float front[3];
+	uint8_t _unused2[4 * 40];
+	/* 9fc */ float pos[3];
+};
+
+#define unreal_to_mumble_units(unreal) (unreal / 100.f)
+
+static_assert(sizeof(struct GroundedCam) == 200, "");
+
+
+mumble_error_t mumble_init(uint32_t) {
+	return MUMBLE_STATUS_OK;
+}
+
+void mumble_shutdown() {
+}
+
+MumbleStringWrapper mumble_getName() {
+	static const char name[] = "Grounded";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = name;
+	wrapper.size           = strlen(name);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+MumbleStringWrapper mumble_getDescription() {
+	static const char description[] = "Positional audio support for Grounded. Steam release version 1.4.3.4578.";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = description;
+	wrapper.size           = strlen(description);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+MumbleStringWrapper mumble_getAuthor() {
+	static const char author[] = "MumbleDevelopers";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data           = author;
+	wrapper.size           = strlen(author);
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+mumble_version_t mumble_getAPIVersion() {
+	return MUMBLE_PLUGIN_API_VERSION;
+}
+
+void mumble_registerAPIFunctions(void *) {
+}
+
+void mumble_releaseResource(const void *) {
+}
+
+mumble_version_t mumble_getVersion() {
+	return { 1, 0, 0 };
+}
+
+uint32_t mumble_getFeatures() {
+	return MUMBLE_FEATURE_POSITIONAL;
+}
+
+uint8_t mumble_initPositionalData(const char *const *programNames, const uint64_t *programPIDs, size_t programCount) {
+	for (size_t i = 0; i < programCount; ++i) {
+		if (strcmp(programNames[i], "Maine-Win64-Shipping.exe") != 0) {

```

Comment:
I would define `Maine-Win...` as a `std::string` constant at the beginning of the function. Then you can simply do a `programNames[i] != myConstant` and also reuse the constant in other places where you need the process name (retyping it every time introduces the potential of typos)